pkg/collector/corechecks/oracle-dbm/statements.go

```diff
@@ -575,204 +576,209 @@ func (c *Check) StatementMetrics() (int, error) {
 				c.fqtEmitted = getFqtEmittedCache()
 			}
 
-			if _, found := c.fqtEmitted.Get(queryRow.QuerySignature); !found {
-				FQTDBMetadata := FQTDBMetadata{Tables: queryRow.Tables, Commands: queryRow.Commands}
-				FQTDB := FQTDB{Instance: c.cdbName, QuerySignature: queryRow.QuerySignature, Statement: SQLStatement, FQTDBMetadata: FQTDBMetadata}
-				FQTDBOracle := FQTDBOracle{
-					CDBName: c.cdbName,
-				}
-				FQTPayload := FQTPayload{
-					Timestamp:    float64(time.Now().UnixMilli()),
-					Host:         c.dbHostname,
-					AgentVersion: c.agentVersion,
-					Source:       common.IntegrationName,
-					Tags:         c.tagsString,
-					DBMType:      "fqt",
-					FQTDB:        FQTDB,
-					FQTDBOracle:  FQTDBOracle,
-				}
-				FQTPayloadBytes, err := json.Marshal(FQTPayload)
-				if err != nil {
-					log.Errorf("%s Error marshalling fqt payload: %s", c.logPrompt, err)
+			if sendFqtAndPlan {
+				if (i+1)%10 == 0 && time.Since(start).Seconds() >= float64(c.config.QueryMetrics.MaxRunTime) {
```

Comment:
> What does (i+1)%10 == 0 exactly do? Limit how frequently we do the comparison? 

Exactly. I didn't want to call the timer with every iteration, as it can be thousands of them. In such cases, it shouldn't matter if we do 9 additional iterations before stopping

> not seeing where we set sendFqtAndPlan to false.

Corrected now, thanks.